gapic-generator/templates/default/service/client/_client.erb

```diff
@@ -97,7 +97,11 @@ class <%= service.client_name %>
     # Create credentials
     credentials = @config.credentials
     <%- unless service.generic_endpoint? -%>
-    credentials ||= Credentials.default scope: @config.scope
+    enable_self_signed_jwt = @config.scope == <%= service.client_name %>.configure.scope &&
+                             @config.endpoint == <%= service.client_name %>.configure.endpoint &&
+                             !@config.endpoint.split(".").first.include?("-")
```

Comment:
I think the reason we check for dash should be explained in comments here, along with (potentially) comparing to "documentai.googleapis.com", per Brent's comment. 
It would be nice to not have to trace this to this PR in the generator from library code.

Reply:
Added a comment.